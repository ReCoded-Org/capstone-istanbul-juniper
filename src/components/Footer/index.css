@import url("https://fonts.googleapis.com/css2?family=Moul&family=Roboto&display=swap");

.footerContainer {
  height: 495px;
  padding-top: 390px;
  background-color: #2a8c79;
  color: white;
  background-image: url(./../../images/footerBg.svg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center 14px;
  position: relative;
  font-family: Roboto;
}

.footerContainer a {
  color: white;
}

.footerContainer__links__nav {
  text-align: center;
}

.footerContainer__links {
  margin-bottom: 20px;
}

a.social {
  margin: 0 1rem;
  transition: transform 250ms;
  display: inline-block;
}

a.social:hover {
  transform: translateY(-2px);
}

a.youtube {
  color: white;
}

a.facebook {
  color: white;
}

img.footerContainer__globe {
  position: absolute;
  top: 46px;
  left: 51px;
  transition: all 1s;
  max-width: 260px;
  min-width: 150px;
}

a.instagram {
  color: white;
}

@media (max-width: 500px) {
  img.footerContainer__globe {
    top: 94px;
    transform: scale(0.7);
  }
}
