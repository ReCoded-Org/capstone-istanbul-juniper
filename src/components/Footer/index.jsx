import React from "react";
import { Row, Col } from "antd";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faYoutube,
  faFacebook,
  faInstagram,
} from "@fortawesome/free-brands-svg-icons";
import "./index.css";
import { Link, useLocation } from "react-router-dom";
import { useTranslation } from "react-i18next";
import globe from "./../../images/earthNav.svg";

const LIGHT_GREEN_BACKGROUND_PAGES = [
  "/about",
  "/resources",
  "/games/memorygame",
  "/games/puzzlegame",
  "/games/quizgame",
  "/login",
];

const LIGHT_GREEN = "#b4d0c4";
const DARK_GREEN = "#2a8c79";

const Footer = () => {
  const curRoute = useLocation().pathname;
  const [t] = useTranslation();

  return (
    <div
      className="footerContainer"
      style={{
        backgroundColor: LIGHT_GREEN_BACKGROUND_PAGES.includes(curRoute)
          ? LIGHT_GREEN
          : DARK_GREEN,
      }}
    >
      <footer>
        <img alt="globe" src={globe} className="footerContainer__globe" />
        <Col>
          <Row
            gutter={[25]}
            justify="center"
            className="footerContainer__links"
          >
            <Col className="footerContainer__links__nav">
              <Link to="/trademark"> {t("footer.trademark")}</Link>
            </Col>
            <Col className="footerContainer__links__nav">
              <Link to="/contact"> {t("footer.contact")}</Link>
            </Col>
            <Col className="footerContainer__links__nav">
              <Link to="/policies"> {t("footer.policies")}</Link>
            </Col>
            <Col className="footerContainer__links__nav">
              <Link to="/games"> {t("footer.games")}</Link>
            </Col>
            <Col className="footerContainer__links__nav">
              <Link to="/about"> {t("footer.about")}</Link>
            </Col>
          </Row>
          <Row justify="center" className="footerContainer__social">
            <Col>
              {/* note: facebook,instagram,youtube are built in fortawesome classes */}
              <a href="https://www.facebook.com" className="facebook social">
                <FontAwesomeIcon icon={faFacebook} size="2x" />
              </a>
            </Col>
            <Col>
              <a href="https://www.instagram.com" className="instagram social">
                <FontAwesomeIcon icon={faInstagram} size="2x" />
              </a>
            </Col>
            <Col>
              <a href="https://www.youtube.com" className="youtube social">
                <FontAwesomeIcon icon={faYoutube} size="2x" />
              </a>
            </Col>
          </Row>
        </Col>
      </footer>
    </div>
  );
};

export default Footer;
