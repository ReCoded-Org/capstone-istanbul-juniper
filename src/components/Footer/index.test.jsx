import React from "react";
import renderer from "react-test-renderer";
import Footer from "./";
import { BrowserRouter as Router } from "react-router-dom";

it("Footer renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Footer />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
