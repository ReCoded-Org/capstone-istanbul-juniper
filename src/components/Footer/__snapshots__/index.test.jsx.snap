// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <img
    alt="globe"
    className="footerGlobe"
    src="/assets/images/earthNav.svg"
  />
  <div
    className="ant-col"
    style={Object {}}
  >
    <div
      className="ant-row ant-row-center footerLinksContainer"
      style={
        Object {
          "marginLeft": -12.5,
          "marginRight": -12.5,
        }
      }
    >
      <div
        className="ant-col footerLinksContainer__nav"
        style={
          Object {
            "paddingLeft": 12.5,
            "paddingRight": 12.5,
          }
        }
      >
        <a
          href="/trademark"
          onClick={[Function]}
        >
           
          footer.trademark
        </a>
      </div>
      <div
        className="ant-col footerLinksContainer__nav"
        style={
          Object {
            "paddingLeft": 12.5,
            "paddingRight": 12.5,
          }
        }
      >
        <a
          href="/contact"
          onClick={[Function]}
        >
           
          footer.contact
        </a>
      </div>
      <div
        className="ant-col footerLinksContainer__nav"
        style={
          Object {
            "paddingLeft": 12.5,
            "paddingRight": 12.5,
          }
        }
      >
        <a
          href="/policies"
          onClick={[Function]}
        >
           
          footer.policies
        </a>
      </div>
      <div
        className="ant-col footerLinksContainer__nav"
        style={
          Object {
            "paddingLeft": 12.5,
            "paddingRight": 12.5,
          }
        }
      >
        <a
          href="/games"
          onClick={[Function]}
        >
           
          footer.games
        </a>
      </div>
      <div
        className="ant-col footerLinksContainer__nav"
        style={
          Object {
            "paddingLeft": 12.5,
            "paddingRight": 12.5,
          }
        }
      >
        <a
          href="/about"
          onClick={[Function]}
        >
           
          footer.about
        </a>
      </div>
    </div>
    <div
      className="ant-row ant-row-center footerSocialContainer"
      style={Object {}}
    >
      <div
        className="ant-col"
        style={Object {}}
      >
        <a
          className="facebook social"
          href="https://www.facebook.com"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook fa-w-16 fa-2x "
            data-icon="facebook"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
      <div
        className="ant-col"
        style={Object {}}
      >
        <a
          className="instagram social"
          href="https://www.instagram.com"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-instagram fa-w-14 fa-2x "
            data-icon="instagram"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
      <div
        className="ant-col"
        style={Object {}}
      >
        <a
          className="youtube social"
          href="https://www.youtube.com"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-youtube fa-w-18 fa-2x "
            data-icon="youtube"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</footer>
`;
